### ISSUE TITLE
"Error: Unable to resolve root class of SWF"  when package has a name

### ISSUE THREAD

**Issue by mintyh** [BowlerHatLLC/vscode-as3mxml#399]
"Error: Unable to resolve root class of *.as" (where *.as is your main as file) seems to fire when your package has any name.

So this is fine:
`package {`

and this is not:
`package packageName {`

I'm wondering if it's related to this ASC2 compiler confirmed bug issue:
https://forums.adobe.com/thread/1270697

**Comment by joshtynjala** [comment]
I believe that you can place your main class in a package, but you also need to set the source-path compiler option.

Assuming that the root of your source folder is *src* and your main class is defined in *Main.as*, your *asconfig.json* file should look something like this:

```json
{
	"compilerOptions": {
		"source-path": [
			"src"
		]
	},
	"files": [
		"src/packageName/Main.as"
	]
}
```

**Comment by mintyh** [comment]
Hi Josh,

Thanks for the suggestion... that's the one in the docs, and I tried it, and it had no effect. I got the same error (much as the OP noted in the ASC2 link above).

I've recreated the entire project using just package {}  (without a package name) and made one edit to asconfig.json to take the package name out of the Main.as path... and it no longer reports that error. At this stage, it does point to a bug with package names, as noted in the adobe forum link.

One note: in my tests, files is relative to source-path, so in your example it'll return "unable to open Main.as"... until files is changed to "packageName/Main.as". Otherwise it's looking for Main.as in "src/src/packageName/"...

**Comment by joshtynjala** [comment]
Please see the attached project for a working example.

[MainInPackage.zip](https://github.com/BowlerHatLLC/vscode-as3mxml/files/3531879/MainInPackage.zip)

**Comment by mintyh** [comment]
Hey Josh, thanks so much for such direct, concrete help.

I've tested that and it works fine here using Harmon's AIR SDK 33.0.1.228 . I've tried using the same options in my project... and I still get the Unable to Resolve error.

It's frustrating. I need to carve more time to work out what the contributing factor is. I will post back when I get the chance to do that; for now, other deadlines are shouting at me.

Again, many thanks for the support!
